docs/ai/tutorials/snippets/evaluate-with-reporting/MyTests.cs

```diff
@@ -0,0 +1,153 @@
+    IChatClient chatClient,
+    string astronomyQuestion)
+    {
+        const string SystemPrompt =
+            """
+        You're an AI assistant that can answer questions related to astronomy.
+        Keep your responses concise and try to stay under 100 words.
+        Use the imperial measurement system for all measurements in your response.
+        """;
+
+        IList<ChatMessage> messages =
+            [
+                new ChatMessage(ChatRole.System, SystemPrompt),
+            new ChatMessage(ChatRole.User, astronomyQuestion)
+            ];
+
+        var chatOptions =
+            new ChatOptions
+            {
+                Temperature = 0.0f,
+                ResponseFormat = ChatResponseFormat.Text
+            };
+
+        ChatResponse completion = await chatClient.GetResponseAsync(messages, chatOptions);
+        return (messages, completion);
+    }
+    // </SnippetGetResponse>
+
+    // <SnippetValidate>
+    /// <summary>
+    /// Runs basic validation on the supplied <see cref="EvaluationResult"/>.
+    /// </summary>
+    private static void Validate(EvaluationResult result)
+    {
+        // Retrieve the score for relevance from the <see cref="EvaluationResult"/>.
+        NumericMetric relevance =
+            result.Get<NumericMetric>(RelevanceTruthAndCompletenessEvaluator.RelevanceMetricName);
+        Assert.IsFalse(relevance.Interpretation!.Failed);
+        Assert.IsTrue(relevance.Interpretation.Rating is EvaluationRating.Good or EvaluationRating.Exceptional);
+
+        // Retrieve the score for truth from the <see cref="EvaluationResult"/>.
+        NumericMetric truth = result.Get<NumericMetric>(RelevanceTruthAndCompletenessEvaluator.TruthMetricName);
+        Assert.IsFalse(truth.Interpretation!.Failed);
+        Assert.IsTrue(truth.Interpretation.Rating is EvaluationRating.Good or EvaluationRating.Exceptional);
+
+        // Retrieve the score for completeness from the <see cref="EvaluationResult"/>.
+        NumericMetric completeness =
+            result.Get<NumericMetric>(RelevanceTruthAndCompletenessEvaluator.CompletenessMetricName);
+        Assert.IsFalse(completeness.Interpretation!.Failed);
+        Assert.IsTrue(completeness.Interpretation.Rating is EvaluationRating.Good or EvaluationRating.Exceptional);
+
+        // Retrieve the word count from the <see cref="EvaluationResult"/>.
+        NumericMetric wordCount = result.Get<NumericMetric>(WordCountEvaluator.WordCountMetricName);
+        Assert.IsFalse(wordCount.Interpretation!.Failed);
+        Assert.IsTrue(wordCount.Interpretation.Rating is EvaluationRating.Good or EvaluationRating.Exceptional);
+        Assert.IsFalse(wordCount.ContainsDiagnostics());
+        Assert.IsTrue(wordCount.Value <= 100);
+    }
+    // </SnippetValidate>
+
+    // <SnippetTestMethod>
+    [TestMethod]
+    public async Task SampleAndEvaluateResponse()
```

Comment:
Consider running 2/3 iterations (as in the `SampleAndEvaluateMultipleResponses*` samples).

Since each response can be different, the typical guidance is to sample and evaluate a few responses to a question. So, it may be a good idea to call this out.

Another idea may be to leave the code as is but include a note around the above in the tutorial text and point to an of the example in the ai-samples repo to show how to sample multiple responses (by specifying an `iterationName` for each of the sampled responses)...

Reply:
I added a note about this under Next Steps.